examples/api/lib/material/app/app.0.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+/// Flutter code sample for [MaterialApp].
+
+void main() {
+  runApp(const MaterialAppExample());
+}
+
+class MaterialAppExample extends StatefulWidget {
+  const MaterialAppExample({super.key});
+
+  @override
+  State<MaterialAppExample> createState() => _MaterialAppExampleState();
+}
+
+class _MaterialAppExampleState extends State<MaterialAppExample> {
+  bool isDarkTheme = false;
+  bool customizedThemeAnimation = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      themeMode: isDarkTheme ? ThemeMode.dark : ThemeMode.light,
+      theme: ThemeData(colorScheme: ColorScheme.fromSeed(
+        seedColor: Colors.green,
+      )),
+      darkTheme: ThemeData(colorScheme: ColorScheme.fromSeed(
+        seedColor: Colors.green,
+        brightness: Brightness.dark,
+      )),
+      // When the button with `Customize Theme Animation` text is pressed,
+      // the theme animation will be overridden with the following
+      // [AnimationOverride] instance in the [MaterialApp].
+      themeAnimationOverride: customizedThemeAnimation ? AnimationOverride(
```

Comment:
A request that comes up now and then is for a way to prevent an animation from happening at all. Perhaps we should provide a constant like AnimationOverride.noAnimation that has the durations set to zero. This feature would be useful to include here, since that's sometimes what developers want to do to the theme animation.

Reply:
Great idea!